### ISSUE TITLE
The u and w update in the SiLIFLayer

### ISSUE THREAD

**Issue by JDYG** [Maxtimer97/SSM-inspired-LIF#1]
Hi,
In the SiLIFLayer, the update for wt and ut uses the varibles `B_u` and `R_u`.  But in your [code](https://github.com/Maxtimer97/SSM-inspired-LIF/blob/fd1ac36d6bdf295dac708747882ce5dc162979f1/models/snns.py#L319), the `B_u` and `R_u` are both defined as `(1-alpha)`. May I ask, why did you define the two varibles with the same value?

```python
        B_u = (1- alpha)
        R_u = (1- alpha)

        # Loop over time axis
        for t in range(Wx.shape[1]):

            wt = beta * wt + a * ut + b * st #+ B_w*Wx[:, t, :] 
            ut = alpha * (ut - st) + B_u* Wx[:, t, :] - R_u * wt

            # Compute spikes with surrogate gradient
            st = self.spike_fct(ut - self.threshold)
            s.append(st)

        return torch.stack(s, dim=1)
```
Besides, according to the equation of AdLIF:

<img width="390" height="105" alt="Image" src="https://github.com/user-attachments/assets/1f15cd3c-b4d5-4433-b25e-978f661e935f" />

The code for calculating the `wt` seems missing `(1-beta)` before the `a*ut`. Do you combine the `(1-beta)` with the parameter `a` intentionally due to they are scale variables?

**Comment by Maxtimer97** [comment]
Hello,

Thank you very much for your interest in the repo and your question (and in fact good catch)!

For the `B_u` and `R_u` intermediate variables, these were reminiscent from intermediate experiments and do not make sense in the current code base. I just pushed a new version without them.

For the model's equations, there is in fact a discrepency between the code and the paper. The equations in the code without `(1 - beta)` are the ones we used for the reported results as they follow the adLIF implementation in the _Sparch_ repository.
We will update the paper soon to match our code and experiments.

Thank you again for your message and let me know if you have further questions.

PS: I also realized the scheduler update after each epoch was missing (deleted when cleaning the repository) and I added it again with the latest push. This should have some impact to reproduce the best results.
